src/object/schema_system_catalog_builder.cpp

```diff
@@ -0,0 +1,270 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "schema_system_catalog_builder.hpp"
+
+// #include "dbi.h" /* db_create_class () */
+#include "db.h"
+#include "dbtype.h"
+// #include "dbtype_function.h" /* DB_IS_NULL () */
+#include "authenticate.h"
+#include "class_object.h" /* SM_TEMPLATE */
+#include "schema_template.h" /* smt_edit_class_mop() */
+#include "schema_manager.h"
+#include "locator_cl.h"
+#include "schema_system_catalog_definition.hpp"
+
+namespace cubschema
+{
+  MOP
+  system_catalog_builder::create_and_mark_system_class (const std::string_view name)
+  {
+    if (name.empty ())
+      {
+	// should not happen
+	assert (false);
+	return nullptr;
+      }
+
+    // class_mop
+    MOP class_mop = nullptr;
+    if (sm_is_system_class (name))
+      {
+	class_mop = db_create_class (name.data ());
+	if (class_mop != nullptr)
+	  {
+	    sm_mark_system_class (class_mop, 1);
+	  }
+      }
+    else if (sm_is_system_vclass (name))
+      {
+	class_mop = db_create_vclass (name.data ());
```

Comment:
After db_init() -- here catalogs are created, By sm_mark_system_classes(), all resident classes are marked as system class. Because of this, When I try to remove `sm_mark_system_class (class_mop, 1);` at line 47 above, It works.

However, as this routine has been migrated from legacy code, my emphasis is on introducing an improved API. I intend to move forward by addressing the removal of redundant functions in a new issue.